acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,94 @@
+
+  step 'On master, create new environments that will result in Failed' do
+    site_manifest = File.join(environmentpath, tmp_environment, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  fail('intended failure')
+}
+SITEPP
+    site_manifest2 = File.join(environmentpath, tmp_environment2, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest2, <<-SITEPP)
+node default {
+  exec{'/bin/false':}
+}
+SITEPP
+  end
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent),
+                         pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  target_identities = []
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment}" do
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'),
+               "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal(tmp_environment, action_result['environment'],
+                     "Result of pxp-module-puppet run on #{agent} should run with the "\
+                     "#{tmp_environment} environment")
+        assert_equal('failed', action_result['status'],
+                     "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: "\
+                     "'#{action_result["status"]}'")
+      end
+    end
+  end
+
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment2}" do
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment2,
```

Comment:
redhat7-64m and redhat7-64a

I didn't complain about beaker's methods creating files with restricted permissions; as I assumed it was done deliberately as security best practice. Maybe it's actually a beaker bug. I suppose ideally beaker should always create files with the same permissions you'd get if you manually ssh'd in and created the file